hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreFileInfo.java

```diff
@@ -270,7 +270,10 @@ public StoreFileReader open(FileSystem fs, CacheConfig cacheConf, boolean canUse
       status = fs.getFileStatus(initialPath);
     }
     long length = status.getLen();
-    hdfsBlocksDistribution = computeHDFSBlocksDistribution(fs);
+    if (shared) {
```

Comment:
I know why we need to test this but it is a bit strange...

Reply:
new PR has refact this